scripts/ci/install-dev-deps

```diff
@@ -25,4 +26,7 @@ def run(command):
 
 if __name__ == "__main__":
     with cd(REPO_ROOT):
+        if sys.version_info[:2] >= (3, 12):
+            run("pip install setuptools")
```

Comment:
Do we want to bound the version of setuptools in this install command? It feels like it would probably be worth it to make sure we are pulling in a consistent version of setuptools for our CI.

Reply:
I asked a similar question for the equivalent PR in botocore https://github.com/boto/botocore/pull/2967#discussion_r1293967757